### PR TITLE
WAZO-777 enable wazo-ui or create an index.html

### PR DESCRIPTION
**Pull request by fredericlepied**

### CODE REVIEW

File: roles/uc-ui/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- name: Install wazo-ui
+  become: true
+  apt:
+    name: wazo-ui
+    state: latest
+
+- name: Configure ui
+  template:
+    src: templates/config.yml.j2
+    dest: /etc/wazo-ui/config.yml
```
Comment (sduthil) [inline roles/uc-ui/tasks/main.yml]: `/etc/wazo-ui/config.yml` should be provided by the Debian package. Additional config should be added in `/etc/wazo-ui/conf.d/*.yml`.

File: zuul.d/run.yaml [hunk 2]
```diff
@@ -22,3 +31,8 @@
       command: "ansible-playbook -i inventories/uc-engine test.yml"
       args:
         chdir: "{{ zuul.project.src_dir }}"
+
+    - name: Check ui service
+      become: yes
+      shell: "systemctl status wazo-ui"
```
Comment (sduthil) [inline zuul.d/run.yaml]: How about:

```
- name: Get service facts
  service_facts:
- name: Assert wazo-ui is running
  assert:
    that: "ansible_facts.services['zuul-gateway'].state == 'running'"
```

Comment (fredericlepied) [reply zuul.d/run.yaml]: done

File: roles/uc-ui/tasks/main.yml (hunk 1 above)
Comment (fredericlepied) [reply roles/uc-ui/tasks/main.yml]: done

**Comment by fredericlepied** [comment]
recheck

File: roles/uc-ui/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- name: Install custom certificate for HTTPS
+  copy:
+    src: "{{ engine_api_https_cert }}"
+    dest: /usr/share/xivo-certs/server.crt
+    owner: root
+    group: www-data
+    mode: 0640
+  when: engine_api_https_cert is defined
+  notify:
+    - restart wazo-ui
```
Comment (fblackburn1) [inline roles/uc-ui/tasks/main.yml:36]: Why notify wazo-ui, when it is not installed?

File: roles/uc-ui/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- name: Install custom private key for HTTPS
+  copy:
+    src: "{{ engine_api_https_private_key }}"
+    dest: /usr/share/xivo-certs/server.key
+    owner: root
+    group: www-data
+    mode: 0640
+  when: engine_api_https_private_key is defined
+  notify:
+    - restart wazo-ui
```
Comment (fblackburn1) [inline roles/uc-ui/tasks/main.yml:48]: same comment as above

File: roles/uc-ui/templates/config.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  listen: {{ uc_ui_listen_address }}
+  port: {{ uc_ui_port}}
+  certificate: /usr/share/xivo-certs/server.crt
+  private_key: /usr/share/xivo-certs/server.key
+
+auth:
+  host: {{ engine_api_host }}
+  port: {{ engine_api_port }}
+  prefix: /api/auth
+  verify_certificate: false
```
Comment (fblackburn1) [inline roles/uc-ui/templates/config.yml.j2]: Do not ship default config with `verify_certificate `to False !

File: roles/uc-ui/templates/config.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+confd:
+  host: {{ engine_api_host }}
+  port: {{ engine_api_port }}
+  prefix: /api/confd
+  verify_certificate: false
+
+policies:
```
Comment (fblackburn1) [inline roles/uc-ui/templates/config.yml.j2]: remove unused section

File: roles/uc-ui/templates/config.yml.j2 [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  # You must set a valid UUID policy
+  reseller:
+  customer:
+  location:
+
+enabled_plugins:
```
Comment (fblackburn1) [inline roles/uc-ui/templates/config.yml.j2]: Since is already the default config, remove it

File: roles/uc-ui/templates/config.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+log_filename: /var/log/wazo-ui.log
+
+# PID file.
+pid_filename: /var/run/wazo-ui/wazo-ui.pid
+
+https:
+  listen: {{ uc_ui_listen_address }}
+  port: {{ uc_ui_port}}
+  certificate: /usr/share/xivo-certs/server.crt
+  private_key: /usr/share/xivo-certs/server.key
```
Comment (fblackburn1) [inline roles/uc-ui/templates/config.yml.j2]: Since it's the same value than the default, remove it

File: roles/uc-ui/templates/config.yml.j2 [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+# Log file.
+log_filename: /var/log/wazo-ui.log
+
+# PID file.
+pid_filename: /var/run/wazo-ui/wazo-ui.pid
+
+https:
+  listen: {{ uc_ui_listen_address }}
+  port: {{ uc_ui_port}}
```
Comment (fblackburn1) [inline roles/uc-ui/templates/config.yml.j2]: I don't know if we want to have custom listen address and port. I such a pain to migrate after ...

File: zuul.d/run.yaml [hunk 10]
```diff
@@ -22,3 +31,11 @@
       command: "ansible-playbook -i inventories/uc-engine test.yml"
       args:
         chdir: "{{ zuul.project.src_dir }}"
+
+    - name: Get service facts
+      become: yes
+      service_facts:
```
Comment (fblackburn1) [inline zuul.d/run.yaml:37]: the empty key is needed?

File: zuul.d/run.yaml [hunk 11]
```diff
@@ -1,6 +1,15 @@
 ---
 - hosts: all
   tasks:
+    - name: Activate uc-ui in inventory
+      become: yes
+      blockinfile:
+        path: "{{ zuul.project.src_dir }}/inventories/uc-engine"
+        marker: "# Uncomment the 2 following lines if you want to deploy wazo-ui too:"
```
Comment (fblackburn1) [inline zuul.d/run.yaml:8]: I don't know what is the good practice, but it seems a little bit weak to have a marker to a comment that doesn't deserve to be a marker

File: roles/uc-ui/defaults/main.yml [hunk 12]
```diff
@@ -0,0 +1,2 @@
+uc_ui_listen_address: 0.0.0.0
+uc_ui_port: 9296
```
Comment (fblackburn1) [inline roles/uc-ui/defaults/main.yml]: Do you have any use case for this?

**Review by fblackburn1 — CHANGES_REQUESTED** [review]
I'm not familiar with ansible, so sorry for some naive questions.
All default wazo-ui configuration should be removed and not duplicated. It is so horrible to migrate after having a drop-in config file.
Also, I'm not sure about the use case to allow custom values for listening address and port. If we do not have use case, I would prefer to avoid it and remove the custom config file completely.

File: zuul.d/run.yaml (hunk 10 above)
Comment (fredericlepied) [reply zuul.d/run.yaml:37]: yes that's the mandatory type of the task

File: roles/uc-ui/defaults/main.yml (hunk 12 above)
Comment (fredericlepied) [reply roles/uc-ui/defaults/main.yml]: No it was just mimicking the other services so removing it for now.

File: roles/uc-ui/templates/config.yml.j2 (hunk 8 above)
Comment (fredericlepied) [reply roles/uc-ui/templates/config.yml.j2]: ok

File: roles/uc-ui/templates/config.yml.j2 (hunk 7 above)
Comment (fredericlepied) [reply roles/uc-ui/templates/config.yml.j2]: ok

File: zuul.d/run.yaml (hunk 11 above)
Comment (fredericlepied) [reply zuul.d/run.yaml:8]: it was to edit like a user would do it...

File: roles/uc-ui/templates/config.yml.j2 (hunk 9 above)
Comment (fredericlepied) [reply roles/uc-ui/templates/config.yml.j2]: ok

File: roles/uc-ui/templates/config.yml.j2 (hunk 6 above)
Comment (fredericlepied) [reply roles/uc-ui/templates/config.yml.j2]: ok

File: roles/uc-ui/tasks/main.yml (hunk 3 above)
Comment (fredericlepied) [reply roles/uc-ui/tasks/main.yml:36]: when we execute uc-ui/tasks/main.yml, wazo-ui is always installed

File: roles/uc-ui/tasks/main.yml (hunk 4 above)
Comment (fredericlepied) [reply roles/uc-ui/tasks/main.yml:48]: sale as above

File: roles/uc-ui/templates/config.yml.j2 (hunk 5 above)
Comment (fredericlepied) [reply roles/uc-ui/templates/config.yml.j2]: we need to discuss this one because it will not work if the ui is on a separate host

**Review by fblackburn1 — APPROVED** [review]

**Pull request merged**